typescript/cli/src/deploy/core.ts

```diff
@@ -62,13 +61,14 @@ export async function runCoreDeploy(params: DeployParams) {
       'Select chain to connect:',
     );
   }
-
   let apiKeys: ChainMap<string> = {};
   if (!skipConfirmation)
     apiKeys = await requestAndSaveApiKeys([chain], chainMetadata, registry);
 
+  const signer = multiProvider.getSigner(chain);
```

Comment:
From what this PR is trying to solve, having a signer in context would be redundant since it can be fetched from multiProvider which itself comes from context.

Reply:
@ltyu @xeno097  any thoughts? If you guys don't have any strong opinions I don't either.